cpp/include/cugraph_c/cugraph_api.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuda_runtime_api.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum cugraph_error_ {
+  CUGRAPH_SUCCESS,
+  CUGRAPH_ERROR_UNKNOWN,
+  CUGRAPH_INVALID_HANDLE
+} cugraph_error_t;
+
+typedef int bool_t;
+
+/* C stub declarations */
+
+typedef struct c_raft_handle_ {
+  int allign_;
+} c_raft_handle_t;
+
+typedef struct c_graph_envelope_ {
+  int allign_;
+} c_graph_envelope_t;
+
+typedef struct c_erased_device_array_ {
+  int allign_;
+} c_device_array_t;
+
+typedef struct c_erased_unique_ptr_ {
+  int allign_;
+} c_unique_ptr_t;
+
+typedef struct c_device_buffer_ {
+  void* data_;
+  size_t size_; /* in bytes */
+} c_device_buffer_t;
+
+/* C algorithm specific stubs: should go into separate corresponding headers */
+
+typedef struct c_rw_ret_tuple_ {
+  c_device_buffer_t vertex_paths_;
+  c_device_buffer_t weight_paths_;
+  c_device_buffer_t sizes_;
+} c_rw_ret_tuple_t;
+
+typedef struct c_rw_ret_ {
+  void* p_erased_ret;
+} c_rw_ret_t;
+
+/* TODO:
+ * (1.) graph_envelope "cnstr" / "destr":
+ *
+ *      c_graph_envelope_t* make_graph_envelope(...);
+ *      free_graph_envelope(c_graph_envelope_t* ptr);
+ *
+ * (2.) type reconstruction extractors;
+ *      e.g., for `return_t`: different extractors
+ *      interpret return in a different (typed) way;
+ *
+ *      Example:
+ *      c_device_buffer_t* extract_rw_ret_vertex_path(c_type vertex_t_id, c_rw_ret_t* rw_result);
+ *      c_device_buffer_t* extract_rw_ret_weight_path(c_type vertex_t_id, c_rw_ret_t* rw_result);
+ */
+
+/* C algorithm specific wrapper declarations: : should go into separate corresponding headers */
+
+cugraph_error_t c_random_walks(const c_raft_handle_t* ptr_handle,
```

Comment:
I think this may have been mentioned before, but now would be a good time to change the naming convention if we want to do that. Within RAPIDS, cuML follows the convention of using the project name as a prefix (as seen [here](https://github.com/rapidsai/cuml/blob/branch-21.10/cpp/include/cuml/cuml_api.h)), as do other CUDA libraries such as [this](https://docs.nvidia.com/cuda/cusolver/index.html#cusolverDNXgesv).  Related to that is the dreaded topic of case style; cuML and the other CUDA libraries use camelCase and not snake_case.

Reply:
I'm not so sure about camelCase. All our classes (types) use snake_case in the cugraph code (and `_t` suffix).

Reply:
I don't have a strong opinion about snake vs. camel, so leaving it snake_case is also fine by me if that's what we want. @ChuckHastings @BradReesWork any thoughts?

Reply:
I'd prefer to keep things as snake_case unless there's a compelling reason to change them.  Personally not a fan of the `_t` as widely as we are using it, but it's OK.

I do think that rather than `c_` as the prefix, making it `cugraph_` would probably be more informative.  The fact that it's a C API is immaterial to the calling user.  The fact that it's part of a cugraph package is far more informative.  It would also reduce the likelihood of name collision, since we would likely be the only source of C functions that start with `cugraph_`.